R/formatter.r

```diff
@@ -767,49 +767,32 @@ number_bytes_format <- function(symbol = "auto", units = "binary", ...) {
 #' @export
 #' @rdname number_bytes_format
 number_bytes <- function(x, symbol = "auto", units = c("binary", "si"), ...) {
+  units <- match.arg(units, c("binary", "si"))
+
   symbols <- c(
-    "auto",
-    "b", "Kb", "Mb", "Gb", "Tb", "Pb", "Eb", "Zb", "Yb",
-    "B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB",
-    "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB"
+    "auto", "B",
+    switch(units,
+      si     = c( "kB",  "MB",  "GB",  "TB",  "PB",  "EB",  "ZB",  "YB"),
+      binary = c("KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB")
+    )
   )
 
   if (!(symbol %in% symbols)) {
-    warning(paste(
-      "`symbol` must be one of '", paste0(symbols, collapse = "', '"),
-      "'. Defaulting to 'auto'."
+    warning(paste0(
```

Comment:
Can you eliminate the `paste()` altogether? `warning()` already pastes together its inputs